internal/ingress/controller/controller.go

```diff
@@ -131,6 +132,19 @@ type Configuration struct {
 	DynamicConfigurationRetries int
 }
 
+func getIngressPodZone(svc *apiv1.Service) string {
+	svcKey := k8s.MetaNamespaceKey(svc)
+	if svcZoneAnnotation, ok := svc.ObjectMeta.GetAnnotations()[apiv1.AnnotationTopologyAwareHints]; ok {
+		if strings.ToLower(svcZoneAnnotation) == "auto" {
+			if foundZone, ok := k8s.IngressNodeDetails.GetLabels()[apiv1.LabelTopologyZone]; ok {
+				klog.V(3).Infof("Svc has topology aware annotation enabled, try to use zone %q where controller pod is running for Service %q ", foundZone, svcKey)
+				return foundZone
+			}
+		}
+	}
+	return emptyZone
```

Comment:
why defining a new var when you can just return empty? I would just do return "" and remove this var declaration

Reply:
It's not defined just for one use, it appears here
https://github.com/kubernetes/ingress-nginx/pull/9165/files#diff-e2c540c6da86368f0fa693d5634db0a92e8ef119e5cc2609bd3d24baf6c0fa8aR116
I defined it for better readability, not just comparing against empty string, but against variable which name give some hint what is going on.

Reply:
Now it has even more usage.